numpyro/contrib/control_flow/cond.py

```diff
@@ -0,0 +1,131 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+from functools import partial
+
+from jax import device_put, lax
+
+from numpyro import handlers
+from numpyro.contrib.control_flow.util import PytreeTrace
+from numpyro.primitives import _PYRO_STACK, apply_stack
+
+
+def _subs_wrapper(subs_map, site):
+    if isinstance(subs_map, dict) and site["name"] in subs_map:
+        return subs_map[site["name"]]
+    elif callable(subs_map):
+        rng_key = site["kwargs"].get("rng_key")
+        subs_map = (
+            handlers.seed(subs_map, rng_seed=rng_key)
+            if rng_key is not None
+            else subs_map
+        )
+        return subs_map(site)
+    return None
+
+
+def wrap_fn(fn, substitute_stack):
+    def wrapper(wrapped_operand):
+        rng_key, operand = wrapped_operand
```

Comment:
In `scan_wrapper` the `rng_key` is split inside `body_fn`. I understood that that was because you are iterating and you don't want to use the same key on each iteration. In this case we just need to pass the single key down to the wrapped function, is that correct?

Reply:
I believe it is correct.